sw/airborne/modules/mission/copilot.c

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (C) 2016  2017 Michal Podhradsky <http://github.com/podhrmic>
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ */
+/**
+ * @file "modules/mission/copilot.c"
+ *
+ *  Mission Computer module, interfacing the mission computer (also known as Copilot),
+ *  based losely on
+ *  ISaAC: The Intelligent Safety and Airworthiness Co-Pilot module
+ *  Based on paper "A Payload Verification and Management Framework
+ *  for Small UAV-based Personal Remote Sensing Systems" by Cal Coopmans
+ *  and Chris Coffin. Link: http://ieeexplore.ieee.org/abstract/document/6309316/
+ *
+ *  More info can be found on http://wiki.paparazziuav.org/wiki/Mission_computer
+ *
+ *  Copilot is intended mainly for mapping applications.
+ *
+ *   This module processes messages from Copilot, and either forwards them to the GCS
+ *  (such as CAMERA_SNAPSHOT or CAMERA_PAYLOAD messages), or responds to them as necessary
+ *  (such as MOVE_WP).
+ *
+ *  The module assumes the source of the messages is trusted (i.e. not authentication besides
+ *  AC_ID check is performed).
+ */
+
+#include "modules/mission/copilot.h"
+#include "subsystems/datalink/telemetry.h"
+#include <string.h>
+
+#include "pprz_mutex.h"
+
+// needed for WP_MOVED confirmation
+#include "firmwares/fixedwing/nav.h"
+#include "subsystems/navigation/common_nav.h"
+#include "math/pprz_geodetic_float.h"
+
+bool send_cam_snapshot;
+bool send_cam_payload;
+bool send_copilot_status;
+
+struct CameraPayload cam_payload;
+struct CameraSnapshot cam_snapshot;
+struct CopilotStatus copilot_status;
+
+PPRZ_MUTEX(copilot_cam_snapshot_mtx);
```

Comment:
There might be threads in the future  - so just added protection and good practice. I would rather have an unused mutex than forget to add it later.